src/System/save_component.rs

```diff
@@ -5,12 +5,30 @@ use crate::ModelObjects::representations::BoolExpression;
 use crate::TransitionSystems::{LocationTuple, TransitionSystemPtr};
 use std::collections::HashMap;
 
-pub fn combine_components(system: &TransitionSystemPtr) -> Component {
+pub enum Reachability {
+    Reachable,
+    All,
+}
+
+use Reachability::*;
+
+pub fn combine_components(system: &TransitionSystemPtr, reachability: Reachability) -> Component {
```

Comment:
The naming of this `reachability` parameter is not too clear. If we rename it to `pruning_strategy` and rename the `Reachability` struct to `Pruning` or  `PruningStrategy`, and then change `All` into `None`.

I also think the usage will be much more clearer e.g. `combine_components(my_system, PruningStrategy::Reachable)`